megamek/src/megamek/common/actions/WeaponAttackAction.java

```diff
@@ -1410,27 +1403,17 @@ private static String toHitIsImpossible(Game game, Entity ae, int attackerId, Ta
 
         // limit large craft to zero net heat and to heat by arc
         final int heatCapacity = ae.getHeatCapacity();
-        if (ae.usesWeaponBays() && (weapon != null) && !weapon.getBayWeapons().isEmpty()) {
+        if (ae.usesWeaponBays() && (weapon != null)) {
             int totalHeat = 0;
 
             // first check to see if there are any usable weapons
-            boolean usable = false;
-            for (WeaponMounted m : weapon.getBayWeapons()) {
-                WeaponType bayWType = m.getType();
-                boolean bayWUsesAmmo = (bayWType.getAmmoType() != AmmoType.T_NA);
-                if (m.canFire()) {
-                    if (bayWUsesAmmo) {
-                        if ((m.getLinked() != null) && (m.getLinked().getUsableShotsLeft() > 0)) {
-                            usable = true;
-                            break;
-                        }
-                    } else {
-                        usable = true;
-                        break;
-                    }
-                }
-            }
-            if (!usable) {
+            if (!weapon.streamBayWeapons()
```

Comment:
Request to keep streams out of if statements. I'd rather have an extra line and boolean assignment and a variable name that says what this test is about.

Reply:
I don't see how having a distinct boolean would add clarity beyond what the existing comment provides.

Reply:
Do you want me to change the comment to be an initialized boolean anyway?

Reply:
I agree with @SJuliez. It makes the code harder to read.